### PR TITLE
Changed generating route only on publish

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | build ontop #122 https://github.com/sulu/sulu/pull/3299
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR changes the time to generate the route to on-publish which allows us to detect deleted nodes and remove their routes.

### CODE REVIEW

File: Controller/ArticlePageController.php [hunk 1]
```diff
@@ -195,16 +195,24 @@ public function putAction($articleUuid, $uuid, Request $request)
...
     {
+        $locale = $this->getRequestParameter($request, 'locale', true);
+
         $documentManager = $this->getDocumentManager();
-        $document = $documentManager->find($uuid);
+        $document = $documentManager->find($uuid, $locale);
         $documentManager->remove($document);
         $documentManager->flush();
 
+        // FIXME this is a current hack which publishes article automatically when removing a single page
```
Comment (wachterjohannes) [inline Controller/ArticlePageController.php]: I will change this in a following PR.

File: composer.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "license": "MIT",
     "require": {
         "php": "^5.5 || ^7.0",
-        "sulu/sulu": "^1.5.2 || dev-develop",
+        "sulu/sulu": "dev-feature/route-generator-pool",
```
Comment (wachterjohannes) [inline composer.json]: will be changed to `dev-develop` after merging the PR sulu/sulu#3299

File: Document/Subscriber/RoutableSubscriber.php [hunk 3]
```diff
@@ -31,8 +36,14 @@
  */
 class RoutableSubscriber implements EventSubscriberInterface
 {
-    const FIELD = 'routePath';
+    const ROUTE_FIELD = 'routePath';
     const TAG_NAME = 'sulu_article.article_route';
+    const ROUTES_FIELD = 'routes';
```
Comment (alexander-schranz) [inline Document/Subscriber/RoutableSubscriber.php]: `ROUTES_PROPERTY` or `PROPERTY_NAME_ROUTES`

Comment (alexander-schranz) [inline Document/Subscriber/RoutableSubscriber.php]: we should also prefix this propery maybe `sulu-routes`

Comment (wachterjohannes) [inline Document/Subscriber/RoutableSubscriber.php]: done (:

**Pull request merged**